org.eclipse.xtext.ui.testing/xtend-gen/org/eclipse/xtext/ui/testing/AbstractHighlightingTest.java

```diff
@@ -178,11 +174,19 @@ protected StyledText openInEditor(final IFile dslFile) {
     }
   }
   
+  /**
+   * @since 2.18
+   */
+  protected void waitForEventProcessing() {
```

Comment:
maybe we should have a util for this somewhere.
see `org.eclipse.xtext.ui.editor.quickfix.TextualMultiModificationWorkbenchMarkerResolutionAdapter.processUIEvents(int)` too

Reply:
Having a utility method is a good idea, but that kind of refactoring I would see in a separate issue/pr as soon as this pr has been approved/merged.

Reply:
I created https://github.com/eclipse/xtext-eclipse/issues/1018 as the follow up ticket for the refactoring.